sonic_platform_base/sonic_xcvr/sfp_optoe_base.py

```diff
@@ -183,6 +183,13 @@ def read_eeprom(self, offset, num_bytes):
         except (OSError, IOError):
             return None
 
+    def read_eeprom_by_page(self, page, offset, size, wire_addr=None, flat=False):
+        api = self.get_xcvr_api()
```

Comment:
Can we check for api == None?
Please do a similar thing in write_eeprom_by_page function too.

Reply:
It already checked api == None in the next line by `if api is not None`:

```
overall_offset = api.get_overall_offset(page, offset, size, wire_addr, flat) if api is not None else None
```

And the same check is already in write_eeprom_by_page